lib/improver/nowcasting/optical_flow.py

```diff
@@ -886,20 +885,21 @@ def process(self, cube1, cube2):
         # calculate plugin parameters in grid square units
         self.data_smoothing_radius = \
             int(self.data_smoothing_radius_km / grid_length_km)
-        self.boxsize = int(self.boxsize_km / grid_length_km)
 
         # Fail verbosely if self.data_smoothing_radius is too small and will
-        # trigger silent failures downstream. (Note that self.boxsize < 3 will
-        # also trigger silent failures, but this is caught indirectly by
-        # enoforcing boxsize > data_smoothing_radius at initialisation.)
+        # trigger silent failures downstream
         if self.data_smoothing_radius < 3:
             msg = ("Input data smoothing radius {} too small (minimum 3 "
                    "grid squares)")
             raise ValueError(msg.format(self.data_smoothing_radius))
 
-        # TODO raise further warnings if data_smoothing_radius or boxsize
-        # have values (to be determined) that are scientifically questionable.
-        # Here if dimensionless; at initialisation if dimensioned.
+        # Fail if self.boxsize is less than self.data_smoothing_radius
+        self.boxsize = boxsize
```

Comment:
This is the only place in the class where `boxsize` is not a class attribute (although it's passed into `_make_subboxes` as that function is hideous without abbreviation!).  We need it to be passed into the `process` function for configurability, to allow the plugin to be tuned if we wanted to run on different grids.  I've tidied up some docstrings which erroneously refer to `boxsize` rather than `self.boxsize` - hopefully this will help?